CHANGELOG.md

```diff
@@ -38,6 +38,31 @@
   }
   ```
 
+- Add support for writing policies in Python :tada:
+  (https://github.com/pulumi/pulumi-policy/pull/212).
+
+  Example:
+
+  ```python
+  def s3_no_public_read(args: ResourceValidationArgs, report_violation: ReportViolation):
+      if args.resource_type == "aws:s3/bucket:Bucket" and "acl" in args.props:
+          acl = args.props["acl"]
+          if acl == "public-read" or acl == "public-read-write":
+              report_violation("You cannot set public-read or public-read-write on an S3 bucket.")
+
+  PolicyPack(
+      name="aws-policy-pack",
+      enforcement_level=EnforcementLevel.MANDATORY,
+      policies=[
+          ResourceValidationPolicy(
+              name="s3-no-public-read",
+              description="Prohibits setting the publicRead or publicReadWrite permission on AWS S3 buckets.",
+              validate=s3_no_public_read,
+          ),
+      ],
+  )
```

Comment:
Worth mentioning: there are two ways to define policies in Python that Luke and I discussed.

First is this way, which is similar to how policies are declared in TypeScript. Unfortunately, lambdas in Python must be expressions and cannot have statements, so it necessitates defining the handler function separately.

The second way involves inheriting from `ResourceValidationPolicy` or `StackValidationPolicy` and overriding the `validate` function. This would look like:

```python
class S3NoPublicRead(ResourceValidationPolicy):
    def __init__(self):
        super().__init__(
            name="s3-no-public-read",
            description="Prohibits setting the publicRead or publicReadWrite permission on AWS S3 buckets.")

    def validate(args, report_violation):
        if args.resource_type == "aws:s3/bucket:Bucket" and "acl" in args.props:
              acl = args.props["acl"]
              if acl == "public-read" or acl == "public-read-write":
                  report_violation("You cannot set public-read or public-read-write on an S3 bucket.")

PolicyPack(
      name="aws-policy-pack",
      enforcement_level=EnforcementLevel.MANDATORY,
      policies=[
          S3NoPublicRead(),
      ],
  )
```

Luke and I also chatted about possibly using Python decorators to automatically add policies with a certain decorator to a Policy Pack, but that is not implemented in this PR (but could potentially be built on-top of this lower-level support). I have not yet had a chance to experiment with this.